oasislmf/model_preparation/summaries.py

```diff
@@ -575,40 +575,33 @@ def write_exposure_summary(
     except OasisException:   # Empty dataframe (due to empty keys errors file)
         gul_inputs_errors_df = pd.DataFrame(columns=gul_inputs_df.columns.append(pd.Index(['status'])))
 
-    def split_dataframe_list(df, target_column, separator):
-        """
-        Split dataframe with multiple perils into separate rows.
-        """
-        row_accumulator = []
-
-        def split_list_to_rows(row, separator):
-            split_row = row[target_column].split(separator)
-            for s in split_row:
-                new_row = row.to_dict()
-                new_row[target_column] = s
-                row_accumulator.append(new_row)
-        df.apply(split_list_to_rows, axis=1, args=(separator, ))
-        new_df = pd.DataFrame(row_accumulator)
-        return new_df
-
     # Merge GUL input items and source exposure dataframes to leave covered
     # perils
     loc_num = oed_hierarchy['locnum']['ProfileElementName'].lower()
     loc_per_cov = oed_hierarchy['locperilid']['ProfileElementName'].lower()
     model_peril_ids = gul_inputs_df['peril_id'].unique()
-    exposure_df = split_dataframe_list(exposure_df, 'locperilscovered', ';')
+    exp_perils_df = pd.DataFrame(
+        exposure_df[loc_per_cov].str.split(';').to_list(),
+        index=exposure_df[loc_num]
+    ).stack()
+    exp_perils_df = exp_perils_df.reset_index([0, loc_num])
```

Comment:
This can be simplified using the `inplace` optional argument as follows

    exp_perils.reset_index([0, loc_num], inplace=True)

Passing `inplace=True` will reset the index on the dataframe directly, without creating a new dataframe which must be reassigned to the original, and will therefore be a bit faster.

Reply:
Changed. Just pushed the change to the repo.

Reply:
That list you're passing to `reset_index` will be processed as the value of the `level` optional argument - that applies when the dataframe has a multi-index with several index levels, and you only want to remove some of these. Do you need that here?

Reply:
For the same reason there is an error in PiWind which is causing the CI build to fail.

Reply:
I can see that `exp_perils_df` has a multi-index. If there is no way of doing this without creating a new dataframe and a reassignment then that's fine. But there should be a way of doing this using `inplace=True` also, which will be more efficient.